### ISSUE TITLE
Retrieving likes for FB posts

### ISSUE THREAD

**Issue by kboghe** [JustAnotherArchivist/snscrape#144]
First of all: thanks for creating and constantly maintaining this package; it must be a madman's job with the frequent code changes from Twitter and (especially) Facebook. 

Anyway, I'm wondering whether I could retrieve the amount of likes/shares a Facebook post has received using the FacebookUserScraper class. Right now it just returns the post and a timestamp. I'm not exactly a Python wizard, so I'm having a hard time figuring out how your code is structured. Any help on this would be greatly appreciated!

**Comment by JustAnotherArchivist** [comment]
Thanks for the kind words. Fortunately, things don't break too frequently, but Facebook is indeed very annoying to work with.

It should be possible to extract that information from what Facebook returns, and I intend to add various other things as well anyway, e.g. photos and videos, comments count, share count, and proper poster information (relevant especially for communities). And the reaction count, of course.
I've been holding off on this so far because Facebook is rolling out a new design, which will almost certainly require large changes in the extraction in general, and I don't have a desire to implement these things twice. That new design was supposed to launch in September, but so far it's only used on group pages.

**Comment by kboghe** [comment]
Fair enough! Best of luck with this project going forward; I'm definitely following this one.

**Comment by hans-ekbrand** [comment]
> It should be possible to extract that information from what Facebook returns, and I intend to add various other things as well anyway, e.g. photos and videos, comments count, share count, and proper poster information (relevant especially for communities). And the reaction count, of course.
> I've been holding off on this so far because Facebook is rolling out a new design, which will almost certainly require large changes in the extraction in general, and I don't have a desire to implement these things twice. That new design was supposed to launch in September, but so far it's only used on group pages.

I also think snscrape is great, I've used it daily for half a year now. Do you still plan to implement fetching comments on posts in groups? That's what I lack the most now.

**Comment by JustAnotherArchivist** [comment]
@hans-ekbrand Once Facebook properly rolls out the redesign, I'll implement post metadata, including the comment and like counts. I've also thought about adding comment retrieval, but there are no concrete plans for this currently.

**Comment by neon-ninja** [comment]
Comment extraction is implemented in https://github.com/kevinzg/facebook-scraper, if that helps

**Comment by JustAnotherArchivist** [comment]
Based on a small sample, it appears that everything uses the new layout now, almost two years after they started rolling it out... So I guess this is finally feasible.
